### ISSUE TITLE
Crash attempting to cast an OpenBlocks paint mixer to an itemduct on every third save

### ISSUE THREAD

**Issue by rockobonaparte** [CoFH/Feedback#1106]
I am having a block ID-related problem in 1.7.10 that probably stems from my own special shenanigans. A log:

https://www.dropbox.com/s/5buf8y1rd5e3umr/fml-client-latest.log?dl=0

It's on Dropbox because it's 3.5MB. It's 3.5MB because it represents starting and stopping a world three times in a row, which is how many times it takes to crash for me. It's very deterministic though.

Most notably, Thermal Dynamics is attempting to cast an OpenBlocks paint mixer to a duct. The link I see is that the mixer comes up as block/item IDs #1370, but Forge reassigns it to ID #1267. The itemduct has the ID #1370.

Important details:
1. It happens on every third time I quit the world in single-player. I will crash about one second in to the title screen.
2. Placing paint mixers and item ducts doesn't really seem to have any bearing on anything. Between the different saves, I notice no differences.
3. It does look like changes do get saved before the crash--or at least enough of them. Blocks I alter in the world are okay.
4. I am using Thermal Dynamics 1.1.0-161, OpenBlocks 1.7.10-1.5.1 (openmodslib 1.7.1-0.9.1), and Forge 1.7.10-10.13.4.1558.

The really, really important detail:
I am doing a really goofy thing with a modified Lockdown that copies a template world into the save when the game is first started. This template is a previous save. The block IDs are from the original save, and will start to have problems when mods are updated or removed. The code does do the block and item ID repair; you can see it in the log.

I can't be entirely sure it's a Thermal Dynamics bug, but I hope you can verify something in it doesn't latch on to ancient block IDs, although I can't really see how it would.

**Comment by XFactHD** [comment]
Does this also happen if you don't do your "special shenanigans"?

**Comment by rockobonaparte** [comment]
What would be a rational experiment to exclude it? Just a regular old Overworld?

**Comment by XFactHD** [comment]
For example.

**Comment by rockobonaparte** [comment]
The most conservative alteration I could think of doing was disabling the flag that had Lockdown overriding the player files. I had added some code to handle that so it would load a tutorial dimension I had created, while allowing the player's overworld settings to take effect. So instead of that stuff, it just copies over the entire template--basically an existing save, and starts off of that.

The world that loaded up seemed to withstand 5 consecutive opens and closes, so I am pretty sure it at least isn't happening then. I haven't been able to sleuth around about what happened with block IDs in that case.

I need a few days to inspect some stuff with an NBT editor. I was using NBT Explorer, but it likes to hang when searching now for some reason.

**Comment by rockobonaparte** [comment]
I played around with it some more. With normal Lockdown code, it still happens. That is the mod that starts the player off in a specific environment--generally for the sake of an HQM pack. It overrides GuiCreateWorld and puts files under a template directory into the place of the newly-created world. It's kind of like using the Create New World button to load a copy of a friend's save.

Since the time I created that template, I had installed and changed some mods, so the block IDs had drifted. Forge is loading up with one map and the save has a different map. They both save their maps, so Forge can reassign them. According to logs in the original text, that looks like it's going on. However, something in thermal dynamics somehow latches on to the old block ID and kicks in after I've saved and left that world instance.

I'm tempted to just remove Thermal Dynamics and friends from one load of the world and just suck up losing all the machines and conduits. However, I am sure that is going to completely garble my quest book, and it will wreck a lot of demonstration machines I am setting up in the tutorial dimension. It also doesn't really assure me that I won't be running into it all again some point later when I update a mod.
